tests/integration/test_dataapi.py

```diff
@@ -59,30 +52,100 @@ def _build_base_url(self):
         """Make a request against Data Api to ensure the SDK works"""
         client = DataApi()
-        res = client.request(self._build_base_url() + endpoint, security, consumer_secret, request,
+
+        test_url = self._build_base_url() + items_endpoint
+
+        print("test_url is", test_url)
+
+        res = client.request(test_url, security, consumer_secret, items_request,
                              action)
-        print(res.request.url)
-        print(res.request.body)
-        print(res.content)
+
         returned_json = res.json()
+
         assert len(returned_json['data']) > 0
+
         returned_ref = returned_json['data'][0]['reference']
-        assert returned_ref in request['references']
+
+        assert returned_ref in items_request['references']
 
     def test_paging(self):
         """Verify that paging works"""
         client = DataApi()
-        pages = client.request_iter(self._build_base_url() + endpoint, security, consumer_secret,
-                                    request, action)
+
+        test_url = self._build_base_url() + items_endpoint
+
+        print("test_url is", test_url)
+
+        pages = client.request_iter(test_url, security, consumer_secret,
+                                    items_request, action)
+
         results = set()
+
         for page in pages:
             if page['data']:
                 results.add(page['data'][0]['reference'])
 
         assert len(results) == 2
         assert results == {'item_2', 'item_3'}
 
+    def test_real_question_request(self):
+        """Make a request against Data Api to ensure the SDK works"""
+        client = DataApi()
+
+        test_url = self._build_base_url() + questions_endpoint
+
+        print("test_url is", test_url)
+
+        questions_request['limit'] = 3
+
+        res = client.request(test_url, security, consumer_secret, questions_request,
+                             action)
+
+        returned_json = res.json()
+
+        assert len(returned_json['data']) > 0
+
+        keys = set()
+
+        for key, value in self.__iteritems(returned_json['data']):
+            keys.add(key)
+
+        assert keys == {'py-sdk-test-2019-1', 'py-sdk-test-2019-2'}
+
+    def test_question_paging(self):
+        """Verify that paging works"""
+        client = DataApi()
+
+        test_url = self._build_base_url() + questions_endpoint
+
+        print("test_url is", test_url)
+
+        pages = client.request_iter(test_url, security, consumer_secret,
+                                    questions_request, action)
+
+        results = []
+
+        for page in pages:
+            if page['data']:
+                results.append(page['data'])
+
+        keys = set()
+
+        for row in results:
+            for key in row.keys():
+                keys.add(key)
+
+        assert len(results) == 3
+        assert keys == {'py-sdk-test-2019-1', 'py-sdk-test-2019-2'}
+
+    @staticmethod
+    def __iteritems(obj, **kwargs):
```

Comment:
Is it necessary to duplicate this function here, or can we reuse it from the original class or create a util?

Reply:
I could make it a public class method in /request/dataapi.py rather than a private one - other option is to add it to the util library with the UUID code... which do you think is best? Another option is to pull in the python futures library by adding it to requirements.txt, maybe that's too overkill?

Reply:
I think just a little util might be the best way to go. 👍

Reply:
Plus that gives us more space to add documentation. Plus it'd be good to add the documentation - including a reference to the Python Futures library so that we can tell what it does and why it's there.